packages/sanity/src/core/form/inputs/arrays/ArrayOfObjectsInput/Grid/GridArrayInput.tsx

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable react/jsx-handler-names */
+import {Card, Stack, Text} from '@sanity/ui'
+import React from 'react'
+import {UploaderResolver} from '../../../../studio/uploads/types'
+import {Item, List} from '../../common/list'
+import {ArrayOfObjectsInputProps, ObjectItem, ObjectItemProps, UploadEvent} from '../../../../types'
+import {DefaultArrayInputFunctions} from '../../common/ArrayFunctions'
+import {ArrayOfObjectsItem} from '../../../../members'
+import {createProtoArrayValue} from '../createProtoArrayValue'
+import {UploadTargetCard} from '../../common/UploadTargetCard'
+import {GridItem} from './GridItem'
+import {ErrorItem} from './ErrorItem'
+
+export interface ArrayInputProps<Item extends ObjectItem> extends ArrayOfObjectsInputProps<Item> {
+  resolveUploader: UploaderResolver
+  onUpload: (event: UploadEvent) => void
+}
+
+export function GridArrayInput<Item extends ObjectItem>(props: ArrayInputProps<Item>) {
+  const {
+    schemaType,
+    onChange,
+    value = [],
```

Comment:
Probably a good idea to move the default value to a constant to prevent hooks that use value downstream to rerun on render?